File: src/main/java/com/embabel/guide/domain/drivine/HasGuideUserData.java
package com.embabel.guide.domain.drivine;

import org.jetbrains.annotations.NotNull;

/**
 * Interface for composed result types that contain a GuideUser.
 * Allows for polymorphic handling of different query result compositions.
 */
public interface HasGuideUserData {

    /**
     * Get the GuideUser from this composed result
     * @return The GuideUser instance
     */
    @NotNull
    GuideUserData guideUserData();
}

File: src/main/kotlin/com/embabel/guide/domain/HasGuideUserData.kt
package com.embabel.guide.domain

/**
 * Interface for composed result types that contain a GuideUser.
 * Allows for polymorphic handling of different query result compositions.
 */
interface HasGuideUserData {

    /**
     * Get the GuideUser from this composed result
     * @return The GuideUser instance
     */
    fun guideUserData(): GuideUserData
}
